### PR TITLE
Refactored test suite

### PR DESCRIPTION
**Pull request by carlcheel**
This PR adds the following to the test suite:

- Tests now run in parallel, which outputs test pass / fail when they happen and will save time on machines with more than 1 CPU core
- Added @types/jest to enable jest typings in VS Code (and other editors)
- Added ESLint to tests to improve quality
- Changed SLS and npm commands to run asynchronously
- Moved each test into it's relevant folder

Resolves issue #109

### CODE REVIEW

File: tests/scripts/setupTests.js [hunk 1]
```diff
@@ -1,2 +1,6 @@
-// ONLY FOR TESTING INTERNALLY IN --TEST_MODE
-console.log("Running 'setupFilesAfterEnv'");
```
Comment (carlcheel) [inline tests/scripts/setupTests.js:2]: Removed this log as was causing noise in tests. Didn't seem required, is it?

Comment (jayair) [reply tests/scripts/setupTests.js:2]: This is actually a hack I have to test if the serverless-bundle is using the config set in the package.json:

```
  "jest": {
    "setupFilesAfterEnv": [
      "<rootDir>/tests/scripts/setupTests.js"
    ]
  },
```

And we want to test the `./bin/scripts.js`:

```
    "test-scripts": "./bin/scripts.js test --TEST_MODE tests/scripts/test.js"
```

So while it's in the same directory as the Jest tests, it's really a way for me to run it manually to ensure that the above script works. I think I'll move it out of the `tests/` directory. That way we won't need to ignore it:

```
    "test": "jest ./tests --no-watchman --testPathIgnorePatterns=\"./tests/scripts\"",
```

I'll have a look at this before merging.

**Comment by jayair** [comment]
@carlcheel Just tried it. This is awesome. That's exactly what I was looking for. Just left a couple of comments. Let me know what you think.

**Comment by jayair** [comment]
Ended up creating a new branch for this by mistake as I was working on it.

https://github.com/AnomalyInnovations/serverless-bundle/pull/112

**Comment by jayair** [comment]
@carlcheel We are pretty much ready to merge. Just let me know if there is a better way to do what I mentioned in the review.

File: tests/scripts/setupTests.js (hunk 1 above)
Comment (carlcheel) [reply tests/scripts/setupTests.js:2]: Awesome

**Comment by carlcheel** [comment]
@jayair Thanks for taking a look. I can't see your review comments, did you submit the review?

**Comment by jayair** [comment]
Oh yeah submitting it now.

File: tests/aliases/aliases.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { runSlsCommand, clearNpmCache, errorRegex } = require("../helpers");
+
+beforeEach(async () => {
```
[the remark this answers is not in the document]
Comment (carlcheel) [reply tests/aliases/aliases.test.js:3]: We can put it in the Jest setup file, however, I think with this approach it's clear what the tests are doing for someone new to the test suite.

**Comment by carlcheel** [comment]
Closing this PR as @jayair has opened #112 which includes these changes.

File: tests/aliases/aliases.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jayair) [reply tests/aliases/aliases.test.js:3]: Yeah that's fair. Let's go with this.

**Pull request closed without merging**